src/nerpa_pipeline/NRPSPredictor_utils/json_handler.py

```diff
@@ -62,14 +64,35 @@ class SVM_entry:
 
         '''
 
-    def __init__(self, domain_prediction):
-        self.angstrom_code = domain_prediction['angstrom_code']
-        self.stachelhaus_seq = domain_prediction['stachelhaus_seq']
-        self.physicochemical_class = domain_prediction['physicochemical_class']
-        self.large_cluster_pred = domain_prediction['large_cluster_pred']
-        self.small_cluster_pred = domain_prediction['small_cluster_pred']
-        self.single_amino_pred = domain_prediction['single_amino_pred']
-        self.uncertain = domain_prediction['uncertain']
+    def __init__(self, domain_prediction, new_version=False):
+        if new_version:
+            try:
+                self.angstrom_code = domain_prediction['aa34']
+                self.stachelhaus_seq = domain_prediction['aa10']
+                self.physicochemical_class = domain_prediction['physiochemical_class']['name']
+                self.large_cluster_pred = [substrate['short'].lower()
+                                           for substrate in domain_prediction['large_cluster']['substrates']]
+                self.small_cluster_pred = [substrate['short'].lower()
+                                           for substrate in domain_prediction['small_cluster']['substrates']]
+                substrates = domain_prediction['single_amino']['substrates']
+                self.single_amino_pred = substrates[0]['short'].lower() if substrates else 'N/A'
+                if domain_prediction['stachelhaus_matches']:
+                    self.stachelhaus_match_count = max(int(stachelhaus_match['aa10_score'] * 10 + 0.1)  # +0.1 to avoid occasional rounding error
```

Comment:
Why not use `round(x)` for this?

Reply:
Just forgot it existed. Sure, I will change it to `round(x)`.